avm/res/cdn/profile/main.bicep

```diff
@@ -105,6 +108,21 @@ var formattedRoleAssignments = [
   })
 ]
 
+var formattedUserAssignedIdentities = reduce(
+  map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }),
+  {},
+  (cur, next) => union(cur, next)
+) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities)
+  ? {
+      type: (managedIdentities.?systemAssigned ?? false)
+        ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned')
```

Comment:
Can you clarify here why If managedIdentities.systemAssigned is true and If managedIdentities.userAssignedResourceIds is not empty, type is set to 'SystemAssigned,UserAssigned' instead of being set to 'SystemAssigned'? Are you considering users would add a managedIdentitiesType that has both the systemAssigned boolean true and a list of userAssignedResourceIds?

Reply:
alltough the usecase where you use a systemAssigned Managed Identity and one or more User-Assigned Managed identities is a very rare usecase, it is technically possible to do so, and it would result in a valid deployment.
i copied this piece of code from the StorageAccount AVM module and i veried this is also possible for the Cdn Profile resource.

Reply:
Hey @gbeaud, good point and when we defined the interface for the UDT we figured exactly what @cmezach wrote above. Users may or may not want to use both, so we did not want to limit them to 'either or' (unless of course the RP does not support it)